src/main/scala/TileResetCtrl.scala

```diff
@@ -0,0 +1,64 @@
+package testchipip
+
+import chisel3._
+import chisel3.experimental.{IO}
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.devices.tilelink._
+import freechips.rocketchip.regmapper._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.util._
+import freechips.rocketchip.tile._
+import freechips.rocketchip.prci.{ClockSinkDomain, ClockGroupIdentityNode}
+
+// initResetHarts: list of hartids which will stay in reset until its reset-ctrl register is cleared
+case class TileResetCtrlParams(initResetHarts: Seq[Int] = Nil, address: BigInt=0x100000, slaveWhere: TLBusWrapperLocation = PBUS)
+case object TileResetCtrlKey extends Field[TileResetCtrlParams](TileResetCtrlParams())
+
+object TLTileResetCtrl {
+  def apply(sys: BaseSubsystem with InstantiatesTiles)(implicit p: Parameters) = {
+    val resetCtrlParams = p(TileResetCtrlKey)
+    val tlbus = sys.locateTLBusWrapper(resetCtrlParams.slaveWhere)
+    val domain = sys { LazyModule(new ClockSinkDomain(name=Some("tile-reset-ctrl"))) }
+    domain.clockNode := tlbus.fixedClockNode
+    val resetCtrl = domain {
+      LazyModule(new TLTileResetCtrl(tlbus.beatBytes, resetCtrlParams, sys.tile_prci_domains))
+    }
+    tlbus.toVariableWidthSlave(Some("tile-reset-ctrl")) { resetCtrl.node := TLBuffer() }
+    resetCtrl.tileResetProviderNode
+  }
+}
+
+class TLTileResetCtrl(w: Int, params: TileResetCtrlParams, tile_prci_domains: Seq[TilePRCIDomain[_]])(implicit p: Parameters) extends LazyModule {
+  val device = new SimpleDevice("tile-reset-ctrl", Nil)
+  val node = TLRegisterNode(Seq(AddressSet(params.address, 4096-1)), device, "reg/control", beatBytes=w)
+  val tileResetProviderNode = ClockGroupIdentityNode()
+
+  lazy val module = new LazyModuleImp(this) {
+    val nTiles = p(TilesLocated(InSubsystem)).size
+    require (nTiles <= 4096 / 8)
+    val r_tile_resets = (0 until nTiles).map({ i =>
+      Module(new AsyncResetRegVec(w=1, init=(if (params.initResetHarts.contains(i)) 1 else 0)))
+    })
+    node.regmap((0 until nTiles).map({ i =>
+      (i*8) -> Seq(RegField.rwReg(1, r_tile_resets(i).io)),
```

Comment:
Why not just pack them tightly?

Reply:
Yeah these can just be byte-packed. 
I don't want to do bit-packing, because that makes it slightly harder to enable/disable specific harts.

Reply:
Ever so slightly but more than 512 harts is unreasonable

Reply:
It is convenient to be able to bring more than one hart of reset with a single write, but it is also annoying to have to do bit manipulation for a fully bit-packed register (not necessarily for single-bits but bit-packing multi-bit registers has proven annoying).
I think just switching them to byte-packed it the best solution.